### PR TITLE
Define block data structure. Resolves issue #4

### PR DESCRIPTION
**Pull request by zp-sd**
* Put block and header records in blocks.hrl

### CODE REVIEW

File: apps/aecore/include/blocks.hrl [hunk 1]
```diff
@@ -0,0 +1,24 @@
+-record(trees, {}).
+-type(trees() :: #trees{}).
+
+-record(block, {
+          height = 0             :: non_neg_integer(),
+          prev_block_hash = <<>> :: binary(),
```
Comment (wagerlabs) [inline apps/aecore/include/blocks.hrl]: Rename to `prev_hash`? There's no other previous hash here.

File: apps/aecore/include/blocks.hrl [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          trees = #trees{}       :: trees(),
+          trees_root_hash = <<>> :: binary(),
```
Comment (wagerlabs) [inline apps/aecore/include/blocks.hrl]: Rename to `root_hash`? There's no other root hash.

File: apps/aecore/include/blocks.hrl [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+          txs = []               :: list(),
+          difficulty = 0         :: non_neg_integer(),
+          nonce = 0              :: non_neg_integer(),
+          time = 0               :: non_neg_integer(),
+          version = 0            :: non_neg_integer()}).
+-type(block() :: #block{}).
+
+-record(header, {
+          height = 0             :: non_neg_integer(),
+          prev_block_hash = <<>> :: binary(),
```
Comment (wagerlabs) [inline apps/aecore/include/blocks.hrl]: Same as above for prev and root hashes.

File: apps/aecore/include/blocks.hrl [hunk 4]
```diff
@@ -0,0 +1,24 @@
+-record(trees, {}).
+-type(trees() :: #trees{}).
+
+-record(block, {
+          height = 0             :: non_neg_integer(),
+          prev_block_hash = <<>> :: binary(),
+          trees = #trees{}       :: trees(),
```
Comment (wagerlabs) [inline apps/aecore/include/blocks.hrl]: What are these trees? Please add a comment, e.g. _%% Merkle tree of transactions_

File: apps/aecore/include/blocks.hrl [hunk 5]
```diff
@@ -0,0 +1,23 @@
+-include("trees.hrl").
+
+-record(block, {
+          height = 0             :: non_neg_integer(),
```
Comment (lucafavatella) [inline apps/aecore/include/blocks.hrl]: height as standalone type would be handy.

**Pull request merged**